code/go/0chain.net/smartcontract/zcnsc/models.go

```diff
@@ -149,40 +149,38 @@ func (mp *MintPayload) GetStringToSign() string {
 	return encryption.Hash(fmt.Sprintf("%v:%v:%v:%v", mp.EthereumTxnID, mp.Amount, mp.Nonce, mp.ReceivingClientID))
 }
 
-func (mp *MintPayload) verifySignatures(state cstate.StateContextI) error {
+func (mp *MintPayload) countValidSignatures(state cstate.StateContextI) int {
```

Comment:
Let's keep the original `verifySignatures()` method, we don't accept any invalid signatures. What we need to do are:
- validate all the signatures are valid
- there's no duplicate signatures
- all signatures number > threshold, which could be done out of this function